java-frontend/src/test/java/org/sonar/java/bytecode/se/MethodLookupTest.java

```diff
@@ -0,0 +1,69 @@
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.bytecode.se;
+
+import com.google.common.collect.Lists;
+import java.io.File;
+import org.junit.Test;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.util.Printer;
+import org.sonar.java.bytecode.loader.SquidClassLoader;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class MethodLookupTest {
+
+  private static final String TESTCLASS = "org.sonar.java.bytecode.cfg.testdata.MethodLookup#";
+  private static final MethodVisitor NOP_VISITOR = new MethodVisitor(Opcodes.ASM5) {};
+
+  SquidClassLoader squidClassLoader = new SquidClassLoader(Lists.newArrayList(new File("target/test-classes"), new File("target/classes")));
+
+  @Test
+  public void lookup_should_contain_throws_declaration() {
+    MethodLookup lookup = MethodLookup.lookup(TESTCLASS + "throwing()V", squidClassLoader, NOP_VISITOR);
+    assertThat(lookup.declaredExceptions).containsExactly("java.io.IOException");
+  }
+
+  @Test
+  public void lookup_method_from_superclass() {
+    MethodLookup lookup = MethodLookup.lookup(TESTCLASS + "methodDefinedInSuperClass()V", squidClassLoader, NOP_VISITOR);
+    assertThat(lookup.declaredExceptions).containsExactly("java.util.NoSuchElementException");
+
+    lookup = MethodLookup.lookup(TESTCLASS + "methodDefinedInSuperClass2()V", squidClassLoader, NOP_VISITOR);
+    assertThat(lookup.declaredExceptions).containsExactly("java.lang.IllegalArgumentException");
+
+    lookup = MethodLookup.lookup(TESTCLASS + "ifaceMethod()V", squidClassLoader, NOP_VISITOR);
+    assertThat(lookup.declaredExceptions).containsExactly("java.io.FileNotFoundException");
+
+    lookup = MethodLookup.lookup(TESTCLASS + "ifaceMethod2()V", squidClassLoader, NOP_VISITOR);
+    assertThat(lookup.declaredExceptions).containsExactly("java.lang.UnsupportedOperationException");
+  }
+
+  @Test
+  public void superclass_first() {
+    MethodLookup lookup = MethodLookup.lookup(TESTCLASS + "superClassFirst()I", squidClassLoader, new MethodVisitor(Opcodes.ASM5) {
+      @Override
+      public void visitInsn(int opcode) {
+        System.out.println(Printer.OPCODES[opcode]);
```

Comment:
Sout and no assertions in this test ? did you forgot to remove it ?

Reply:
indeed, this test case was an experiment I forgot to remove, thx for spotting it